src/native/minipal/utf8.c

```diff
@@ -532,7 +532,7 @@ static size_t GetCharCount(UTF8Encoding* self, unsigned char* bytes, size_t coun
 
     EncodeChar:
 
-        availableBytes = pEnd - pSrc;
+        availableBytes = (int)(pEnd - pSrc);
```

Comment:
These down-casts can lead to silent integer overflows that are often a security bugs.

It would be better to fix these warnings by changing the types of these variables to size_t.